src/consensus/encode.rs

```diff
@@ -167,7 +167,7 @@ pub fn deserialize<T: Decodable>(data: &[u8]) -> Result<T, Error> {
 /// doesn't consume the entire vector.
 pub fn deserialize_partial<T: Decodable>(data: &[u8]) -> Result<(T, usize), Error> {
     let mut decoder = Cursor::new(data);
-    let rv = Decodable::consensus_decode(&mut decoder)?;
+    let rv = Decodable::consensus_decode_from_finite_reader(&mut decoder)?;
```

Comment:
In 3a87775a22f2957f8b71ca8a68dfaadd5615a3f0:

I'm a liiitle tempted to change the above line to `Cursor::new(&data[..MAX_VEC_SIZE])` but maybe that's a little too weird/surprising.

Reply:
I was thinking about it. It would give a little bit of protection if someone somehow passed a huge vector in here. But then - if the caller allows malicious input to create huge vectors, they kind of already lost anyway. And the ability of decoding/encoding huge data might be useful for someone that can trust it. Eg. `Vec<Block>` stored in the file system or something. And arbitrary limits might get in the way.